Build a Custom Docker Image Updates (#788)
astronomer/docs: astro/develop-project.md

#### Step 3: Build a Custom Docker Image

> 1. Run the following command to create a new Docker image from your `Dockerfile.build` file. Replace the pip repository and associated credentials values with your own.

Reviewer: This should not have changed

Author: Corrected. Copy and paste errors.

    ```sh

>     DOCKER_BUILDKIT=1 docker build -f Dockerfile.build --progress=plain --ssh=github="$HOME/.ssh/<ssh-key>" -t custom-<astro-runtime-image>.

Reviewer 2: The dot here doesn't indicate the end of the line, but actually refers to the path for the "Docker context", i.e. the directory to include for the `docker build` ([docs](https://docs.docker.com/engine/reference/commandline/build/#build-with-path)). Hence, the whitespace is required.

```suggestion
    DOCKER_BUILDKIT=1 docker build -f Dockerfile.build --progress=plain --ssh=github="$HOME/.ssh/<ssh-key>" -t custom-<astro-runtime-image> .
```

>     DOCKER_BUILDKIT=1 docker build -f Dockerfile.build --progress=plain --build-arg PIP_EXTRA_INDEX_URL=https://${<repo-username>}:${<repo-password>}@<private-pypi-repo-domain-name> -t custom-<airflow-image> .

Reviewer: This should not have changed

Author: Corrected

    ```

>     For example, if you have `quay.io/astronomer/astro-runtime:5.0.0` in your `Dockerfile.build`, this command would be:

Reviewer: This should not have changed

Author: Corrected.

    ```sh

>     DOCKER_BUILDKIT=1 docker build -f Dockerfile.build --progress=plain --ssh=github="$HOME/.ssh/<authorized-key>" -t custom-astro-runtime-5.0.0-base.

Reviewer 2: Same here
```suggestion
    DOCKER_BUILDKIT=1 docker build -f Dockerfile.build --progress=plain --ssh=github="$HOME/.ssh/<authorized-key>" -t custom-astro-runtime-5.0.0-base .
```

Author: Thanks @BasPH!

>     DOCKER_BUILDKIT=1 docker build -f Dockerfile.build --progress=plain --build-arg PIP_EXTRA_INDEX_URL=https://${<repo-username>}:${<repo-password>}@<private-pypi-repo-domain-name> -t custom-astro-runtime-5.0.0 .

Reviewer: This should not have changed

    ```

2. Update the following entry in the  `Dockerfile`:

   ```
   FROM custom-astro-runtime:5.0.0
   ```
  :::info

>     Astro runtime base images are built on the `linux/amd64` architecture. You must add the architecture name to the `FROM` statement when your operating system and the image architecture are different. For example, this is the format for the Apple M1 architecture:

Reviewer 2: Same nitpick, but it's the machine architecture, not OS architecture
```suggestion
    Astro runtime base images are built on the `linux/amd64` architecture. You must add the architecture name to the `FROM` statement when your machine and the image architecture are different. For example, this is the format for the Apple M1 architecture:
```

Author: I'll use computer instead of machine as that is the recommended standard. Thanks!

>     Astro runtime base images are built on the `linux/amd64` architecture. You must add the architecture name to the `FROM` statement when your operating system and the image architecture are different. For example, this is the format for the Apple M1 architecture:

Author: @stuart23 can you confirm the original text I wrote is correct? If not, could you clarify your request? It's not clear from your comment what should be changed in the text.

>     Astro runtime base images are built on the `linux/amd64` architecture. You must add the architecture name to the `FROM` statement when your operating system and the image architecture are different. For example, this is the format for the Apple M1 architecture:

Reviewer: Yep Ron, you are correct

    DOCKER_BUILDKIT=1 docker build -f Dockerfile.build --progress=plain --ssh=github="$HOME/.ssh/<authorized-key>" -t custom-astro-runtime-5.0.0-base .
    ```

2. Update the following entry in the  `Dockerfile`:

   ```
   FROM custom-astro-runtime:5.0.0
   ```
  :::info

>     Astro runtime base images are built on the `linux/amd64` architecture. You must add the architecture name to the `FROM` statement when your computer and the image architecture are different. For example, this is the format for the Apple M1 architecture:

Reviewer 3: ```suggestion
    Astro runtime base images are built on AMD64 architecture. You must add the architecture name to the `FROM` statement when your computer and the image architecture are different. For example, this is the format for the Apple M1 architecture:
```

#### Step 3: Build a Custom Docker Image

> 1. Run the following command to create a new Docker image from your `Dockerfile.build` file. Replace the pip repository and associated credentials values with your own.

Reviewer 3: ```suggestion
1. Run the following command to create a new Docker image from your `Dockerfile.build` file. Replace the pip repository and associated credential values with your own.
```

    ```

> 2. Update the following entry in the  `Dockerfile`::

Reviewer 2: ```suggestion
2. Update the following entry in the  `Dockerfile`:
```

   ```
   FROM custom-<airflow-image>
   ```
        :::tip

      Astro runtime base images are built on the `linux/amd64` architecture. You must add the architecture name to the `FROM` statement when your operating system and the image architecture are different. For example, this is the format for the Apple M1 architecture:

            ```
          FROM --platform=linux/amd64 astrobase:latest
          ```

>         :::

Reviewer 2: I wanted to check the alignment in the preview because there's a different number whitespaces used, but somehow the preview isn't displaying these changes?

Reviewer 2: Also, leaving "custom-astro-runtime" here would stay in sync with the other code samples.

Reviewer 2: I found there's 2 sections ("Private GitHub Repo" and "Private PyPi Index"), and this only makes changes in the private PyPi index section. Please update the "Private GitHub Repo" section with the same changes, because the same applies there.

Reviewer 2: Alignment is messing up the formatting:

![image](https://user-images.githubusercontent.com/6249654/173622241-7ec8f5fc-9832-486e-ab21-1486cdd23edb.png)

See https://deploy-preview-788--gallant-neumann-56599d.netlify.app/astro/develop-project#install-python-packages-from-private-sources -> select "Private PyPi Index" tab

Author: @BasPH all of the issues you have identified have been corrected. Thanks!

    ```sh
    DOCKER_BUILDKIT=1 docker build -f Dockerfile.build --progress=plain --build-arg PIP_EXTRA_INDEX_URL=https://${<repo-username>}:${<repo-password>}@<private-pypi-repo-domain-name> -t custom-astro-runtime-5.0.0 .
    ```
2. Update the following entry in the  `Dockerfile`:

  ```
   FROM custom-astro-runtime:5.0.0
   ```
  :::info

>     Astro runtime base images are built on the `linux/amd64` architecture. You must add the architecture name to the `FROM` statement when your ocomputer and the image architecture are different. For example, this is the format for the Apple M1 architecture:

Reviewer 2: Small typo, otherwise LGTM!

```suggestion
    Astro runtime base images are built on the `linux/amd64` architecture. You must add the architecture name to the `FROM` statement when your computer and the image architecture are different. For example, this is the format for the Apple M1 architecture:
```

>     Astro runtime base images are built on the `linux/amd64` architecture. You must add the architecture name to the `FROM` statement when your operating system and the image architecture are different. For example, this is the format for the Apple M1 architecture:

Author: @jwitz the current text is accurate, so I won't be implementing your edit suggestion.

>     Astro runtime base images are built on the `linux/amd64` architecture. You must add the architecture name to the `FROM` statement when your operating system and the image architecture are different. For example, this is the format for the Apple M1 architecture:

Reviewer 2: Same here
```suggestion
    Astro runtime base images are built on the `linux/amd64` architecture. You must add the architecture name to the `FROM` statement when your machine and the image architecture are different. For example, this is the format for the Apple M1 architecture:
```

Author: See previous comment. Thanks!

    ```sh
    DOCKER_BUILDKIT=1 docker build -f Dockerfile.build --progress=plain --build-arg PIP_EXTRA_INDEX_URL=https://${<repo-username>}:${<repo-password>}@<private-pypi-repo-domain-name> -t custom-astro-runtime-5.0.0 .
    ```
2. Update the following entry in the  `Dockerfile`:

   ```
   FROM custom-astro-runtime:5.0.0
   ```
    :::tip

      Astro runtime base images are built on the `linux/amd64` architecture. You must add the architecture name to the `FROM` statement when your operating system and the image architecture are different. For example, this is the format for the Apple M1 architecture:

   ```

>     FROM --platform=linux/amd64 astrobase:latest

Reviewer: This should be `FROM --platform=linux/amd64 custom-astro-runtime:5.0.0`
